tensorflow/core/kernels/segment_reduction_ops_gpu.cu.cc

```diff
@@ -126,6 +193,73 @@ __global__ void UnsortedSegmentCustomKernel(
   }
 }
 
+// SetSegmentFreq counts the frequency of occurrence of each element in
+// 'segment_ids', and stores the result in each corresponding position
+// of 'segment_freq'.
+template <typename Index, int OuterDimTileSize>
```

Comment:
I'm not sure if `OuterDimTileSize` makes sense here, `segment_ids` is a vector right?